userspace/libscap/engine/gvisor/runsc.cpp

```diff
@@ -30,36 +26,37 @@ namespace runsc {
 
 result runsc(char *argv[]) {
 	result res;
-	int pipefds[2];
 
-	int ret = pipe(pipefds);
-	if(ret) {
-		return res;
+	std::string full_command;
+	int i = 0;
+	while(true) {
+		if(argv[i] == nullptr) {
+			break;
+		}
+		full_command.append(argv[i]);
+		full_command.append(" ");
+		i++;
 	}
-
-	pid_t pid = vfork();
-	if(pid > 0) {
-		int status;
-
-		close(pipefds[1]);
-		wait(&status);
+	FILE *cmd_out = popen(full_command.c_str(), "r");
```

Comment:
is it possible to call `runsc` directly without going through `sh -c`? This is what `popen` does AFAICT. It did before via `execvp`

Reply:
Oh i will double check, thanks!

Reply:
Oh wait, which `sh -c`? i can't find it :laughing:

Reply:
https://man7.org/linux/man-pages/man3/popen.3.html

>       The command argument is a pointer to a null-terminated string
>       containing a shell command line.  This command is passed to
>       /bin/sh using the -c flag; interpretation, if any, is performed
>       by the shell.

Reply:
Oh thanks, given the man page entry, i don't think it is feasible to skip the `sh -c` for popen.
Is this a problem?

Reply:
Ok i pushed a version that does not rely upon popen.
See the new example test below: https://github.com/falcosecurity/libs/pull/2037#issuecomment-2363390743

Reply:
Thank you Fede